File: Domain/ListingAggregate/Objects/DetailedDescription.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Domain.ListingAggregate.Enums;
using Microsoft.EntityFrameworkCore;

namespace Domain.ListingAggregate.Objects
{
    public class DetailedDescription
    {
        public int Id { get; set; }
        public string Heading { get; set; }
        public string Text { get; set; }
        public double Length { get; set; }
        public double Width { get; set; }
        public UnitOfLength Unit { get; set; }
        public double GetArea() {
            return Length * Width;
        }
        public Guid ListingId { get; set; }
        public Listing Listing { get; set; }
    }
}

File: client-app/src/app/model/ListingAggregate/Objects/DetailedDescription.ts
import { Listing } from './../Listing';

export interface DetailedDescription {
    id: string;
    index: number;
    heading: string;
    dimensions: Dimensions;
    text: string;
    listingId: string;
    listing: Listing;
}

export interface Dimensions {
    length: number;
    width: number;
    unit: UnitOfLength;
    area: number;
}

export enum UnitOfLength {
    feet,
    metres
}